src/main/java/duke/tasklist/TaskManager.java

```diff
@@ -0,0 +1,235 @@
+                    printTaskAddedConfirmation(taskAdded);
+                }
+            } else {
+                throw new DukeException(INCORRECT_DEADLINE_COMMAND);
+            }
+        } else {
+            throw new DukeException(INCORRECT_DEADLINE_COMMAND);
+        }
+    }
+
+    /**
+     * Adds a task to the task list as a "To do" type task
+     *
+     * @param taskList Task type arraylist to store all the tasks entered by the user
+     * @param userInput takes input from the keyboard
+     * @param isPrintingPreviousList boolean to check whether it is printing from the stored list to determine whether
+     *                               to print the message that task has been successfully added. If true, then no
+     *                               message would be printed as task was already previously added
+     * @throws DukeException if there is an invalid/no task entered in the user input
+     */
+    public static void addTaskAsToDo(ArrayList<Task> taskList, String userInput, Boolean isPrintingPreviousList) throws DukeException {
+        String task = userInput.replace("to do ", "").trim();
+        Task taskAdded = new Todo(task);
+        if (!task.equals("")) {
+            taskList.add(taskAdded);
+            if(!isPrintingPreviousList) {
+                printTaskAddedConfirmation(taskAdded);
+            }
+        } else {
+            throw new DukeException(INCORRECT_TASK_COMMAND);
+        }
+    }
+
+    /**
+     * Deletes a task from the task list
+     *
+     * @param taskList Task type arraylist to store all the tasks entered by the user
+     * @param userInput takes input from the keyboard
+     * @throws DukeException if there is an invalid task number entered in the user input
+     */
+    public static void deleteTaskFromToDo(ArrayList<Task> taskList, String userInput) throws DukeException {
+        int initialWordIndex = 0;
+        int updateWordIndex = 0;
+
+        boolean numberExists = false;
+        String[] splitTask = userInput.replaceAll("[\\p{Alpha}, [\\p{Punct}&&[^-]]+]", " ").trim().split(" ");
+        for (String word : splitTask) {
+            if (GeneralMethods.isValidNumber(word)) {
+                numberExists = true;
+                int finalWordIndex = initialWordIndex - updateWordIndex;
+                int taskNumber = (Integer.parseInt(splitTask[finalWordIndex])) - 1;
+                try {
+                    int originalTaskNumber = taskNumber + updateWordIndex;
+                    printTaskDeletedConfirmation(taskList.get(taskNumber), originalTaskNumber);
+                    taskList.remove(taskNumber);
+                    updateWordIndex ++;
+                } catch (IndexOutOfBoundsException e) {
+                    System.out.println(INCORRECT_DELETE_COMMAND);
+                }
+            }
+            initialWordIndex++;
+        }
+        if (!numberExists) {
+            throw new DukeException(INCORRECT_DELETE_COMMAND);
+        }
+    }
```

Comment:
you can definitely improve SLAP on this method. 
Hint: list out what all operations is this method handling. Then separate them out as much as possible.